feat: disable triggerBigSync

laggingreflex opened the pull request:
Lets you add an option `"triggerBigSync": false` in project conf to disable doing a big rsync.

fixes #79

joelgriffith commented on src/local/index.js:
```diff
@@ -80,15 +80,17 @@ function startProject(config, projectConf) {
-        'encryption'
-      );
-    });
+    if (projectConf.triggerBigSync !== false) {
```
Can probably just do `if (projectConf.triggerBigSync) {`. Older configurations won't have this definition, and will return`undefined`, which is falsy.

laggingreflex replied:
I'm checking for not `false`. If older config has it `undefined` then this would be true which is the current default behavior. `if (projectConf.triggerBigSync) {` would disable it for all older configs that have it undefined.

laggingreflex replied:
Should it better be labeled as something like disableRsync?

laggingreflex commented:
I've updated it to be a `--disable-rsync` cli arg and `disableRsync` as project config

I've also update the `--no-delete`  cli arg I added earlier to be `--disable-deletion` cli arg and `disableDeleteion` project arg

laggingreflex commented:
I'm currently testing it out and having some issues. 

Don't merge it yet, I'll let you know when. Sorry for hasty commits.

laggingreflex commented:
I was forgetting about a particular callback which was causing the issue. Wrapping `triggerBigSync` in an `if` just like that was not a great idea:

```
if(..) {
  triggerBigSync(projectConf, { debug: config.debug }, () => {
    localLog(text.SYNC_ON_LARGE_CHANGE_DONE);
    fsHelper.watch();
  });
}
```
There's a callback there which wouldn't get called.

I've used async/await now (added a babel-preset-stage-0) with which it could be solved and simplified:

```
if (!(config.disableRsync || projectConf.disableRsync)) {
  await triggerBigSync(projectConf, { debug: config.debug });
}
localLog(text.SYNC_ON_LARGE_CHANGE_DONE);
fsHelper.watch();
```
Although it alters the API a bit so a lot of tests are failing. I'll see if I can work on those later in the week as well. Feel free to review now.

joelgriffith commented on src/big-sync.js:
```diff
@@ -43,5 +43,11 @@ export function bigSync(project) {
     rsync.output(consoleLogFromBuffer, consoleLogFromBuffer);
   }
 
```
nice!

joelgriffith commented:
This looks great, we'll see what the CI finds test wise.

joelgriffith commented:
I just pushed some updates to get tests running (properly) again, sorry for the thrash. I can update your PR to resolve the conflicts  here on Friday.
